pkg/operations/upgradecluster.go

```diff
@@ -1,52 +1,73 @@
 package operations
 
 import (
-	"github.com/Azure/acs-engine/pkg/api/vlabs"
-	"github.com/satori/uuid"
+	"fmt"
+
+	"strings"
+
+	"github.com/Azure/acs-engine/pkg/api"
+	"github.com/Azure/acs-engine/pkg/operations/armhelpers"
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/prometheus/common/log"
```

Comment:
I guess the long term is to pull what's already in the RP anyway, and I guess it's doing the same thing the prometheus code is doing.